src/aleph/vm/orchestrator/tasks.py

```diff
@@ -127,3 +134,46 @@ async def stop_watch_for_messages_task(app: web.Application):
         await app["messages_listener"]
     except asyncio.CancelledError:
         logger.debug("Task messages_listener is cancelled now")
+
+
+async def monitor_payments(app: web.Application):
+    logger.debug("Monitoring balances")
+    pool: VmPool = app["vm_pool"]
+    while True:
+        await asyncio.sleep(settings.PAYMENT_MONITOR_INTERVAL)
+
+        # Check if the balance held in the wallet is sufficient holder tier resources
+        for sender, executions in pool.get_executions_by_sender(payment=PaymentType.hold):
+            balance = await get_balance(sender)
+
+            # Stop executions until the required balance is reached
+            required_balance = await get_required_balance(executions)
```

Comment:
This mechanism is not valid for hold tiers, because it only takes into account the executions on that CRN, but means that some user with some ALEPH tokens can allocate different VMs on the different CRNs with the less amount of tokens.

Reply:
How? This does mean that either the scheduler or CCN is currently setting the used ALEPH balance for each user?

Reply:
Even if so, this check here is still a good sanity check, just-in-case.